hw/ip/otbn/dv/otbnsim/sim/wsr.py

```diff
@@ -888,6 +940,43 @@ def read(self, num_bytes: int) -> bytes:
         return ret
 
 
+class KmacPartialWriteWSR(WSR):
+    '''Keccak partial write WSR
+
+    This register defines the numbeer of valid bytes in the
+    KmacMsgWSR register for writing partial words and STRB
+    on the KMAC Req AppIntf
+    '''
+    def __init__(self, name: str, kmac: KmacBlock):
+        super().__init__(name)
+        self._kmac = kmac
+        self._value = None
+        self._next_value = None
+
+    def read_unsigned(self) -> int:
+        return self._value
+
+    def read_mask(self) -> int:
+        if self._value is None:
+            return 32
+        return self._value.bit_count()
```

Comment:
It might make sense to raise an error here if the value isn't a valid mask, just so that software errors are caught as soon as possible. For example:
```suggestion
    def read_mask(self) -> int:
        if self._value is None:
            return 32
        elif self._value != (1 << self._value.bit_count()) - 1:
            raise ValueError(f'Invalid mask for KMAC_PARTIAL_WRITE: {self._value:#066x}')
        return self._value.bit_count()
```